BasicMusicPlayer/app/src/main/java/com/example/basicmusicplayer/MainActivity.kt

```diff
@@ -0,0 +1,92 @@
+package com.example.basicmusicplayer
+
+import android.media.AudioManager
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.webkit.WebResourceRequest
+import android.webkit.WebView
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import android.webkit.WebViewClient
+import android.media.MediaPlayer
+import android.media.AudioAttributes
+import android.provider.MediaStore.Audio
+import android.widget.ProgressBar
+import android.view.View
+import android.util.Log
+import android.widget.Button
+import android.widget.Toast
+import java.io.IOException
+
+//define main activity class and define properties
+class MainActivity : AppCompatActivity() {
+    private lateinit var loadingIndicator: ProgressBar;
+    private lateinit var btnPlayAudio : Button;
+    private lateinit var btnPauseAudio : Button;
+    private var mediaPlayer : MediaPlayer? = null;
+
+
+
+    // initializes the activity and sets the layout
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        // initialization of properties
+        loadingIndicator = findViewById(R.id.loading_indicator)
+        loadingIndicator.visibility = View.INVISIBLE
+        btnPauseAudio = findViewById(R.id.btnPauseAudio)
+        btnPlayAudio = findViewById(R.id.btnPlayAudio)
+
+        // listens for play button to be clicked. if clicked, playAudio() is executed
+        btnPlayAudio.setOnClickListener {
+            playAudio()
+        }
+        // listens for play button to be clicked. if clicked, pauseAudio() is executed
+        btnPauseAudio.setOnClickListener {
+            pauseAudio()
+        }
+    }
+
+    private fun playAudio() {
+        val wxycURL = "http://audio-mp3.ibiblio.org:8000/wxyc-alt.mp3"
+
+        mediaPlayer = MediaPlayer()
```

Comment:
You don't need to reinitialize the player every time you start and stop playing.